lib/axiom/fs/stream/extension_streams.js

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AxiomError from 'axiom/core/error';
+import GenericStreams from 'axiom/fs/stream/generic_streams';
+import PostMessageStreams from 'axiom/fs/stream/post_message_streams';
+import EventWithCallback from 'axiom/fs/stream/generic_streams';
+
+/**
+ * Extend PostMessageStreams for use with a Chrome extension.
+ *
+ * @constructor
+ * @extends {PostMessageStreams}
+ */
+export var ExtensionStreams = function() {
+  PostMessageStreams.call(this);
+
+  /** @type {?string} */
+  this.appId = null;
+};
+
+ExtensionStreams.prototype = Object.create(PostMessageStreams.prototype);
+
+export default ExtensionStreams;
+
+/**
+ * Open an ExtensionStream at given appId
+ *
+ * @param {string} appId The appId where receiving ExtensionStream lives.
+ * @return {Promise}
+ */
+ExtensionStreams.prototype.openExtension = function(appId) {
+  this.appId = appId;
+
+  var port = chrome.runtime.connect(appId);
+  return PostMessageStreams.prototype.open.call(this, {port: port, alreadyConnected: false});
```

Comment:
Well, ExtensionStream isa  PostMessageStream.  Someone could conceivably connect to a port directly if they wanted to, with an ExtensionStream instance.  Also, this would change the signature of the method (passing it an string / appId instead of a Port)